mtupper/homework.md

```diff
@@ -0,0 +1,292 @@
+> **true** - comparing `"numbers"`
+> ```javascript
+> var x = 4;
+> (typeof (x + 1)); // is same as (5) and returns "number"
+> (typeof x); // is same as 4 and also returns "number"
+> ```
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+> **true** - since `(x*1 == x)` evaluates to true, second statement is optional (or `||`), it returns true without needing to evaluate the second statement
+
+**r)** `(x=(typeof (x+(typeof x))))==x`
+> **true** - the statement on the left with always result in `"string"` first and then compare it to itself
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.<br />
+> ```javascript
+> var avg = (x+y+z)/3;
+> ```
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+> ```javascript
+> var avg = (x+y+z)/3,
+>     x += (avg-x)/2,
+>     y += (avg-y)/2,
+>     z += (avg-z)/2;
+> ```
+
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b),
+>     a = w * h;
+> ```
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b);
+>
+> if ( h > w );
+> ```
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b);
+>
+> if ( h > w ) {
+>     r = (w / 2);
+> }
+> else {
+>     r = (h / 2);
+> };
+>
+> C = (2 * Math.PI * r);
+> ```
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b);
+>
+> r = Math.sqrt( Math.pow((w / 2), 2) + Math.pow((h / 2), 2) );
+>
+> C = (2 * Math.PI * r);
+> ```
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+> ```javascript
+> var l,
+>     r,
+>     t,
+>     b;
+>
```

Comment:
if you are going to define a lot of variables in one line its more common to do it this way var a,b,c,d,e,f; unless you are also filling them with info, then it would be. var x=2, 
                                                                          y=4,
                                                                          z= "pickles";